app/chitter.rb

```diff
@@ -0,0 +1,105 @@
+require 'sinatra/base'
+require_relative 'data_mapper_setup'
+require_relative 'helper/current_user'
+require 'sinatra/flash'
+
+class Chitter < Sinatra::Base
+  include Helpers
+
+  use Rack::MethodOverride
+
+  register Sinatra::Flash
+  register Sinatra::Partial
+
+  enable :sessions, :static
+  enable :partial_underscores
+
+  set :session_secret, 'super secret'
+  set :partial_template_engine, :erb
+
+  get '/' do
+    erb :'index'
+  end
+
+  get '/users/new' do
+    @user = User.new
+    erb :'users/new'
+  end
+
+  post '/users' do
+    @user = User.new(name: params[:name],
+                     username: params[:username],
+                     email: params[:email],
+                     password: params[:password],
+                     password_confirmation: params[:password_confirmation])
```

Comment:
There is a way to tidy this up with a permitted params constant and some methods to check the params against them